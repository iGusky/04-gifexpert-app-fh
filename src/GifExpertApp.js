import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) =>{

  // const categorias = ['Evangelion', 'Sailor Moon', 'Star Wars'];

  const [categorias, setCategorias] = useState(defaultCategories);

  //const handleAdd = () => {
    //Primera forma de agregarlo
   // setCategorias ([...categorias, 'HunterXHunter']);
    //Forma alternativa
    //setCategorias( cats => [--cats, 'HunterXHunter']);
  //}

  return (<>
    <h2>GifExpertApp</h2>
    <AddCategory setCategorias = { setCategorias } />
    <hr></hr>

    <ol>
      {
        categorias.map( categoria => {
          // return <li key={ categoria }>{ categoria }</li>
          return <GifGrid 
            key = { categoria }
            category={ categoria }/>
        })
      }
    </ol>

    {/* <button onClick={handleAdd}>agregar</button> */}
  </>);
};

export default GifExpertApp;