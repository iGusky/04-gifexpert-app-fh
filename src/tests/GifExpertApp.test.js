import GifExpertApp from '../GifExpertApp';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

describe('Pruebas del componente <GifExpertApp />', () => {
    test('Debe renderizarse correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorias', () => {
        const categories = ['Evangelion','Star wars'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>)

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
    
    
})
