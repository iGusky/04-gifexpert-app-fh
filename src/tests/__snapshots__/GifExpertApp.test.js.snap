// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Evangelion"
      key="Evangelion"
    />
    <GifGrid
      category="Star wars"
      key="Star wars"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas del componente <GifExpertApp /> Debe renderizarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
